TypeSystem/Schema/schema.h

```diff
@@ -179,12 +179,18 @@ struct CurrentStructPrinter<CPPLanguageSelector::CurrentStructs> {
   static void PrintCurrentStruct(std::ostream& os,
                                  const ReflectedType_Struct& s,
                                  std::function<std::string(TypeID, const std::string&)> type_name) {
-    const std::string struct_name = s.CanonicalName();
+    const std::string struct_name = s.native_name;
     os << "CURRENT_STRUCT(" << struct_name;
     if (s.super_id != TypeID::CurrentStruct) {
       os << ", " << type_name(s.super_id, "");
     }
     os << ") {\n";
+    if (Exists(s.template_id)) {
+      CURRENT_ASSERT(s.native_name.length() >= 3u);
+      CURRENT_ASSERT(s.native_name.substr(s.native_name.length() - 2u) == "_Z");
+      os << "  CURRENT_EXPORTED_TEMPLATED_STRUCT(" << s.native_name.substr(0, s.native_name.length() - 2) << ", "
```

Comment:
This piece should be reused from a single place. This is the third copy of the logic so far.